pkg/cmd/kind/test/internal/testcmd/testcmd.go

```diff
@@ -0,0 +1,363 @@
+		config.Nodes = append(config.Nodes, v1alpha4.Node{
+			Role:  v1alpha4.WorkerRole,
+			Image: flags.ImageName,
+		})
+	}
+
+	// create the cluster
+	logger.V(0).Infof("Creating cluster %q ...\n", flags.Name)
+	if err := provider.Create(
+		flags.Name,
+		cluster.CreateWithV1Alpha4Config(config),
+		cluster.CreateWithRetain(true),
+		cluster.CreateWithWaitForReady(time.Minute),
+		cluster.CreateWithKubeconfigPath(flags.kubeconfigPath()),
+		cluster.CreateWithDisplayUsage(false),
+		cluster.CreateWithDisplaySalutation(false),
+	); err != nil {
+		if errs := errors.Errors(err); errs != nil {
+			for _, problem := range errs {
+				logger.Errorf("%v", problem)
+			}
+			return errors.New("aborting due to invalid configuration")
+		}
+		return errors.Wrap(err, "failed to create cluster")
+	}
+
+	// In Kubernetes CI we don't have real IPv6 connectivity so we need to
+	// employ some work arounds when testing IPv6 clusters
+	// We do this all environments for consistency.
+	if config.Networking.IPFamily == v1alpha4.IPv6Family {
+		if err := enableIPv6Workarounds(logger, provider, flags.Name); err != nil {
+			return err
+		}
+	}
+
+	// run tests
+	skip := details.Skip
+	if details.Parallel {
+		skip = strings.Join([]string{`\[Serial\]`, skip}, "|")
+	}
+	cmd := exec.Command(
+		"./hack/ginkgo-e2e.sh",
+		// don't use the horrid e2e.test """provider""" concept
+		"--provider=skeleton",
+		// need to inform tests of worker count
+		"--num-nodes=", strconv.Itoa(numNodesForE2E),
+		// write test results into artifacts
+		"--report-dir", flags.Artifacts,
+		// we do our own log dump
+		"--disable-log-dump=true",
+		// plumb ginkgo options
+		"--ginkgo.focus", details.Focus,
+		"--ginkgo.skip", skip,
+	).SetEnv(
+		// setting this env prevents ginkgo e2e from trying to run provider setup
+		"KUBERNETES_CONFORMANCE_TEST=y",
+		// setting these is required to make RuntimeClass tests work ... :/
+		"KUBE_CONTAINER_RUNTIME=remote",
+		"KUBE_CONTAINER_RUNTIME_ENDPOINT=unix:///run/containerd/containerd.sock",
+		"KUBE_CONTAINER_RUNTIME_NAME=containerd",
+		// ensure the test uses our kubeconfig
+		"KUBECONFIG="+flags.kubeconfigPath(),
+		"GINKGO_PARALLEL="+boolToYorN(details.Parallel),
+	)
+	return exec.InheritOutput(cmd).Run()
+}
+
+func boolToYorN(y bool) string {
+	if y {
+		return "y"
+	}
+	return "n"
+}
+
+func enableIPv6Workarounds(logger log.Logger, provider *cluster.Provider, name string) error {
+	// get a control plane node to use run kubectl from
+	nodes, err := provider.ListInternalNodes(name)
+	if err != nil {
+		return errors.Wrap(err, "failed to get nodes for coreDNS IPv6 CI fixes")
+	}
+	nodes, err = nodeutils.ControlPlaneNodes(nodes)
+	if err != nil {
+		return errors.Wrap(err, "failed to get control-plane node for coreDNS IPv6 CI fixes")
+	}
+	if len(nodes) < 1 {
+		return errors.New("failed to locate a kind node for coreDNS IPv6 CI fixes")
+	}
+	node := nodes[0]
+
+	// "fix" coreDNS configmap
+	// TODO: refactor this to not be a shell script.
```

Comment:
+1000
This can also be a good /help wanted issue

Reply:
maybe? probably not. the replacements need to be equivalent. i'm not sure how many casual contributors know sed / shell and go string manipulation deeply 🤷‍♂